### PR TITLE
ISPN-10262 Deprecate custom interceptors configuration

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-10262

ISPN-10262 Deprecate custom interceptors configuration

* Deprecate the custom interceptors configuration
* Warn at startup if custom interceptors are present
* Deprecate AdvancedCache.getAsyncInterceptorChain() 
* Replace interceptors configuration with direct AsyncInterceptorChain
  manipulation in all the modules.

### CODE REVIEW

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java [hunk 1]
```diff
@@ -78,9 +66,6 @@ public ProtobufMetadataManagerImpl() {
    @Inject
    protected void init(EmbeddedCacheManager cacheManager, InternalCacheRegistry internalCacheRegistry) {
```
Comment (ryanemerson) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java]: `internalCacheRegistry` no longer needed.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java [hunk 2]
```diff
@@ -225,4 +241,35 @@ public void cacheStarted(ComponentRegistry cr, String cacheName) {
    public void cacheManagerStopped(GlobalComponentRegistry gcr) {
       mbeanServer = null;
    }
+
+
+   private void registerProtobufMetadataCache(InternalCacheRegistry internalCacheRegistry,
```
Comment (ryanemerson) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: This method is never actually called.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java (hunk 1 above)
Comment (ryanemerson) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java]: Can we just `@Inject EmbeddedCacheManager cacheManager` as a class variable?

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java (hunk 2 above)
Comment (danberindei) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: Oops :)

**Comment by ryanemerson** [comment]
@danberindei JCache seems to be blowing up

**Comment by danberindei** [comment]
Fixed @ryanemerson. The server was not starting because I replaced uses of the custom interceptors configuration with direct manipulation of AsyncInterceptorChain and forgot to register the interceptors in the component registry.

**Pull request merged**